### PR TITLE
CXP-1229:  Added endpoint to retrieve list of criterion

### PR DESCRIPTION
**Pull request by RiffVentura**
Added an endpoint to retrieve relevant data from a catalog within which there is a list of product selection criteria

### CODE REVIEW

File: components/catalogs/back/src/Infrastructure/Persistence/FindOneCatalogByIdQuery.php [hunk 1]
```diff
@@ -46,6 +48,8 @@ public function execute(string $id): ?Catalog
             $row['name'],
             $row['owner_username'],
             (bool) $row['is_enabled'],
+            /** @var array<ProductSelectionCriterion> $criteria */
```
Comment (RiffVentura) [inline components/catalogs/back/src/Infrastructure/Persistence/FindOneCatalogByIdQuery.php:51]: Removes info warning from phpsalm
```
INFO: MixedArgument - src/Infrastructure/Persistence/FindOneCatalogByIdQuery.php:50:13 - Argument 5 of Akeneo\Catalogs\ServiceAPI\Model\Catalog::__construct cannot be mixed, expecting array<array-key, array{field: string, operator: string, value: mixed}> (see https://psalm.dev/030)
            \json_decode($row['product_selection_criteria'], true),
```

File: components/catalogs/back/src/ServiceAPI/Model/Catalog.php [hunk 2]
```diff
@@ -9,14 +9,20 @@
...
+    /**
+     * @param array<ProductSelectionCriterion> $productSelectionCriteria
+     */
     public function __construct(
         private string $id,
         private string $name,
         private string $ownerUsername,
         private bool $enabled,
+        /** @var array<ProductSelectionCriterion> $productSelectionCriteria */
+        private array $productSelectionCriteria,
```
Comment (willy-ahva) [inline components/catalogs/back/src/ServiceAPI/Model/Catalog.php:25]: Didn't we say that the product selection should be extracted from the Catalog of the Service API?

File: components/catalogs/back/src/Infrastructure/Persistence/FindCatalogProductSelectionCriteriaQuery.php [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+class FindCatalogProductSelectionCriteriaQuery implements FindCatalogProductSelectionCriteriaQueryInterface
+{
+    public function __construct(
+        private Connection $connection,
+    ) {
+    }
+
+    /**
+     * @inheritDoc
+     * @psalm-suppress MixedReturnTypeCoercion
```
Comment (RiffVentura) [inline components/catalogs/back/src/Infrastructure/Persistence/FindCatalogProductSelectionCriteriaQuery.php:24]: couldn't get rid off :
```

INFO: MixedReturnTypeCoercion - src/Infrastructure/Persistence/FindCatalogProductSelectionCriteriaQuery.php:25:42 - The declared return type 'array<array-key, array{field: string, operator: string, value?: mixed}>|null' for Akeneo\Catalogs\Infrastructure\Persistence\FindCatalogProductSelectionCriteriaQuery::execute is more specific than the inferred return type 'array<array-key, mixed>|null' (see https://psalm.dev/197)
    public function execute(string $id): ?array


INFO: MixedReturnTypeCoercion - src/Infrastructure/Persistence/FindCatalogProductSelectionCriteriaQuery.php:48:16 - The type 'array<array-key, mixed>' is more general than the declared return type 'array<array-key, array{field: string, operator: string, value?: mixed}>|null' for Akeneo\Catalogs\Infrastructure\Persistence\FindCatalogProductSelectionCriteriaQuery::execute (see https://psalm.dev/197)
        return $criteria;
```

File: components/catalogs/back/tests/Integration/Infrastructure/Controller/Internal/GetCatalogDataActionTest.php [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+        $response = $client->getResponse();
+        $payload = \json_decode($response->getContent(), true);
+
+        Assert::assertEquals(200, $response->getStatusCode());
+
+        Assert::assertEquals([
+            'product_selection_criteria' => [
+                [
+                    'field' => 'status',
```
Comment (masacc) [inline components/catalogs/back/tests/Integration/Infrastructure/Controller/Internal/GetCatalogDataActionTest.php:46]:
```suggestion
                    'field' => 'enabled',
```

File: components/catalogs/back/tests/Integration/Infrastructure/Controller/Internal/GetCatalogDataActionTest.php [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+                    'value' => true,
+                ],
+            ],
+        ], $payload);
+    }
+
+    public function testItGetsNotFoundResponseWithWrongId(): void
+    {
+        $client = $this->getAuthenticatedInternalApiClient();
+        $this->createCatalog('ed30425c-d9cf-468b-8bc7-fa346f41dd07', 'Store FR', 'admin');
```
Comment (masacc) [inline components/catalogs/back/tests/Integration/Infrastructure/Controller/Internal/GetCatalogDataActionTest.php:57]: you can remove this line, it's not useful in this test

File: components/catalogs/back/tests/Integration/Infrastructure/Persistence/FindCatalogProductSelectionCriteriaQueryTest.php [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    {
+        $id = 'db1079b6-f397-4a6a-bae4-8658e64ad47c';
+        $this->createUser('owner');
+        $this->createCatalog($id, 'Store US', 'owner');
+
+        $result = $this->query->execute($id);
+
+        $expectedCriteria = [
+            [
+                'field' => 'status',
```
Comment (masacc) [inline components/catalogs/back/tests/Integration/Infrastructure/Persistence/FindCatalogProductSelectionCriteriaQueryTest.php:47]:
```suggestion
                'field' => 'enabled',
```

**Review by masacc — APPROVED** [review]

File: components/catalogs/back/tests/Integration/Infrastructure/Controller/Internal/GetCatalogDataActionTest.php (hunk 5 above)
Comment (RiffVentura) [reply components/catalogs/back/tests/Integration/Infrastructure/Controller/Internal/GetCatalogDataActionTest.php:57]: I've left it on purpose to eliminate potential test bias

**Pull request merged**